File: packages/sdk-ruby/lib/bidstack_crm/resources/deals.rb
require_relative "base"

module BidstackCrm
  module Resources
    class Deals < Base
      def list(page: 1, per_page: 50)
        get("/deals", params: { page: page, perPage: per_page })
      end

      def find(deal_id) = get("/deals/#{deal_id}")
      def create(data, idempotency_key: nil) = post("/deals", body: data, idempotency_key: idempotency_key)
      def update(deal_id, data) = patch("/deals/#{deal_id}", body: data)
      def destroy(deal_id) = delete("/deals/#{deal_id}")
    end
  end
end

File: packages/sdk-python/bidstack_crm/resources/deals.py
"""Deals resource."""
from __future__ import annotations
from typing import Any
from ._base import BaseResource


class DealsResource(BaseResource):
    def list(self, *, page: int = 1, per_page: int = 50) -> dict[str, Any]:
        return self._get("/deals", params={"page": page, "perPage": per_page})

    def get(self, deal_id: str) -> dict[str, Any]:
        return self._get(f"/deals/{deal_id}")

    def create(self, data: dict[str, Any], idempotency_key: str | None = None) -> dict[str, Any]:
        return self._post("/deals", json=data, idempotency_key=idempotency_key)

    def update(self, deal_id: str, data: dict[str, Any]) -> dict[str, Any]:
        return self._patch(f"/deals/{deal_id}", json=data)

    def delete(self, deal_id: str) -> None:
        self._delete(f"/deals/{deal_id}")
